### PR TITLE
Don't always calc activity blocks

### PR DESCRIPTION
**Pull request by ChrisLovering**
https://github.com/python-discord/bot/pull/1971 must be merged before this.

We only truly care about how many activity blocks a user has when they have a small number of messages, as we only use this for the voice gate.

If a user has more than say 1k messages, then we really don't need to calculate their activity blocks, as it's quite an expensive query with many messages.

Also, for the tests we only actually care that thee key:value pairs are equal, the order of them isn't actually important.

The naming of `assertCountEqual` is a little misleading, since it actually tests that the first sequence contains the same elements as second, regardless of their order. See https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

### CODE REVIEW

**Review by jchristgit — APPROVED** [review]
LGTM.

I added the "do not merge" label to prevent somebody from merging it before the bot PR is merged.

**Review by Akarys42 — APPROVED** [review]

**Pull request merged**
